torch_geometric/nn/models/neural_fingerprint.py

```diff
@@ -0,0 +1,48 @@
+import torch
+import torch.nn.functional as F
+from torch import tensor
+
+from torch_geometric.nn.conv import MFConv
+from torch_geometric.nn.dense import Linear
+
+
+class NeuralFingerprint(torch.nn.Module):
+    r"""
+    The Nueral Fingerprint model from the
+    `"Convolutional Networks on Graphs for Learning Molecular Fingerprints"
+    <https://arxiv.org/pdf/1509.09292.pdf>`_ paper to generate fingerprints
+    of molecules.
+
+    Args:
+        num_features (int) : The number of features of each node/atom.
+        fingerprint_length (int) : The length of fingerprint vector required.
+        num_layers (int) : Number of layers in the model (Radius in the paper).
+    """
+    def __init__(self, num_features, fingerprint_length, num_layers):
+        super().__init__()
+        self.num_features = num_features
+        self.fingerprint_length = fingerprint_length
+        self.num_layers = num_layers
+        self.layers = torch.nn.ModuleList()
+        for i in range(self.num_layers):
+            self.layers.append(
+                MFConv(in_channels=self.num_features,
+                       out_channels=self.num_features))
+            self.layers.append(
+                Linear(in_channels=self.num_features,
+                       out_channels=self.fingerprint_length))
+
+    def forward(self, x, edge_index):
```

Comment:
```suggestion
    def forward(self, x: torch.Tensor, edge_index: torch.Tensor) -> torch.Tensor:
```

Reply:
Please also add `SparseTensor` support for the `edge_index` input.

Reply:
I am not sure if I understand what you are implying, could you please elaborate

Reply:
Since `MFConv` supports both `torch.sparse.tensor` and `torch_sparse.SparseTensor`, this method should also support them. You might take a look at [MFConv](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/conv/mf_conv.py).

Reply:
@EdisonLeeeee I request you to please review the PR once again.

Reply:
Tests for SparseTensor need to be added, see also [test_mf_conv](https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_mf_conv.py). Overall, LGTM :)

Reply:
And don't forget the changelog :)

Reply:
I have added test for SparseTensor and also modified the Changelog.